src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -173,3 +173,110 @@ def watch():
         t = threading.Thread(target=watch, daemon=True)
         t.start()
         t.join()
+
+
+class ZFSSnapshot(Service):
```

Comment:
I am thinking of ZFSSnapshot could be of type CRUD, since we can list, create and delete them (perhaps even update?).

What do you think of making it so and implementing usual do_* methods?
That would need to fabricate an unique identifier to a snapshot (using dataset@name), and passing all arguments to do_create via a dict.

Thoughts?